### PR TITLE
ci: add release and docker workflows

### LINKED ISSUE #2: Release CI
**Describe the feature you would like implemented, use as much detail as you can.**
Setup CI to automatically create releases on v* tags. Also setup a docker container to create an image on releases and push to master and dev branches.

- [X] I have checked to see if the same (or similar) feature requests exist
**Generally, if a request already exists, then you should comment on that one, but if you don't think this would work, link the feature request(s) below**

- [ ] I have tried alternative solutions
**If yes, list the solutions you have tried**

- [ ] I have ideas about how to implement this feature
**If yes, list them below**

- [X] I would be willing to help implement this feature
**If yes, great! Provide preferred contact information below if you feel comfortable, if not, we will attempt to reach out to you.**
**You can also submit a pull request at anytime without contacting us first, especially if you think you can implement it all on your own.**

- [ ] I have additional context/comments
**If yes, list them below**

### PR DESCRIPTION
**Pull request by Baconing**
Make CI workflows to automatically create release on tagged push matching `v*`.

Push docker images on release and on push to `master` or `develop` branches.

Also a couple accidental files that were in the wrong branch.

Closes #2 

## Summary by CodeRabbit

- **New Features**
  - Implemented a new trigger for Docker builds upon release events.
  - Added multi-stage build process in Dockerfile for optimized production images.
  - Introduced a new "beeapi" service in the example Docker Compose configuration.

- **Enhancements**
  - Streamlined configuration initialization and access across the application.
  - Updated GitHub Actions workflows for more precise build and release processes.

- **Refactor**
  - Renamed and updated functions in `config` and `database` packages for better clarity and usage.
  - Modified the router initialization to accommodate different application environments.

- **Documentation**
  - Updated example service configuration to guide deployment using Docker Compose.

- **Chores**
  - Removed unnecessary logging statement in the main application file.

### CODE REVIEW

**Comment by Baconing** [comment]
@coderabbitai review please :pleading_face:

**Comment by Baconing** [comment]
@coderabbitai review

File: config/config.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ func GenerateConfig() {
 		panic("Created config file. Please edit it and restart the application.")
 	}
 
-	err = yaml.NewDecoder(configFile).Decode(Configuration)
+	err = yaml.NewDecoder(configFile).Decode(conf)
 
 	if err != nil {
 		panic(err)
```
[the remark this answers is not in the document]
Comment (Baconing) [reply config/config.go:44]: Related to #4 and also an issue for error handling I should make soon.

File: database/database.go [hunk 2]
```diff
@@ -19,11 +19,11 @@ var mdls = []interface{}{
...
-	cfg.Passwd = config.Configuration.Database.Password
+	cfg.User = config.GetConfig().Database.User
+	cfg.Passwd = config.GetConfig().Database.Password
 	cfg.Net = "tcp"
-	cfg.Addr = config.Configuration.Database.Host + ":" + strconv.Itoa(config.Configuration.Database.Port)
-	cfg.DBName = config.Configuration.Database.Name
+	cfg.Addr = config.GetConfig().Database.Host + ":" + strconv.Itoa(config.GetConfig().Database.Port)
+	cfg.DBName = config.GetConfig().Database.Name
 	cfg.ParseTime = true
 	cfg.MultiStatements = true
```
[the remark this answers is not in the document]
Comment (Baconing) [reply database/database.go:29]: For another issue

File: config/config.go [hunk 3]
```diff
@@ -69,3 +69,7 @@ func createConfigFile() {
...
 	}
 }
```
[the remark this answers is not in the document]
Comment (Baconing) [reply config/config.go:71]: #4 and a to be created issue

[the remark this answers is not in the document]
Comment (Baconing) [reply config/config.go:71]: In another conversation I talked about creating an issue about error handling and #4 is about setting up proper logging application-wide.

Point is that it's not an issue in this PR

**Review by Trip-kun — APPROVED** [review]
Lgtm add more documentation to new code to match other code :)

Also make a go style guide or something. There's probably nothing wrong but writing up the standard would be good

**Pull request merged**